packages/services/api/src/modules/preflight-script/module.graphql.ts

```diff
@@ -0,0 +1,53 @@
+import { gql } from 'graphql-modules';
+
+export const typeDefs = gql`
+  type PreflightScript {
+    id: ID!
+    sourceCode: String!
+    createdAt: DateTime!
+    updatedAt: DateTime!
+    # TODO: should I add this field?
+    # createdBy: User!
+  }
+
+  input CreatePreflightScriptInput {
+    sourceCode: String!
+  }
+
+  input UpdatePreflightScriptInput {
+    id: ID!
+    sourceCode: String!
+  }
+
+  extend type Mutation {
+    createPreflightScript(
+      selector: TargetSelectorInput!
+      input: CreatePreflightScriptInput!
+    ): PreflightScriptResult!
+    updatePreflightScript(
+      selector: TargetSelectorInput!
+      input: UpdatePreflightScriptInput!
+    ): PreflightScriptResult!
+  }
+
+  """
+  @oneOf
+  """
+  type PreflightScriptResult {
+    ok: PreflightScriptOkPayload
+    error: PreflightScriptError
```

Comment:
can I just use `interface Error` here?
```suggestion
    error: Error
```